rules/no-unnecessary-polyfills.js

```diff
@@ -0,0 +1,143 @@
+'use strict';
+const semver = require('semver');
+const readPkgUp = require('read-pkg-up');
+const {data: compatTable} = require('core-js-compat');
```

Comment:
This is not enough. Most of the features depend on some modules. You could check [`entries`](https://unpkg.com/core-js-compat@3.21.0/entries.json). For example, if you wanna polyfill stable ECMAScript `Set`, you load, for example, `core-js/es/set`. This entry contains those modules:

![image](https://user-images.githubusercontent.com/2213682/152694876-9d1dee3f-0e2f-406f-a36e-da2861cd4b45.png)

so should be checked data for all of those modules.

Reply:
If I understood correctly, that's what this regex does in a more general way:
https://github.com/sindresorhus/eslint-plugin-unicorn/blob/6eba19eac46aad8ec166f6a95118b6c1914681be/rules/no-unnecessary-polyfills.js#L54-L63

Reply:
Yes, I understand. But `data` is about modules, each feature/entry can contain some modules, for the proper result, it should handle data for all required modules. Your approach makes sense, but you can get this data by related entry, not module, name. For example, `es6-array-copy-within` / `array-copy-within` / etc -> `core-js/stable/array/copy-within` and check all required modules.

Reply:
I got that idea. But the reason why I preferred to keep this solution is that it's more general and it's not reserved to core-js. Unless I do two checks, one with the regex and one with `entries` but I think it would be overkill.

Reply:
I don't think that two such checks are overkill. The problem is that `@babel/compat-data` is oriented to `core-js@2` and incomplete, `core-js-compat` - to `core-js@3`, and we can't get the proper generic data for all possible polyfills from any of those sources. (For one project, I have a plan to add to `core-js-compat` a list of mapping (like `entries`) of popular third-party polyfills -> similar `core-js` modules, but it will be not very soon.)

The most proper way is usage `core-js-compat` for `core-js` and data from `compat-table` and `caniuse` for polyfills that haven't such data, but it's really overkilling.

Reply:
> If I understand correctly your approach is (to be sure):

Yes, something like that.